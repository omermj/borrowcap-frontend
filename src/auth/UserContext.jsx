import React from "react";

/** Context to provide currentUser thoughout the app */

const UserContext = React.createContext();

export default UserContext;
