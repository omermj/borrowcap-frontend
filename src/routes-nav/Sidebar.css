.Sidebar a  {
  color: white;
}

.Sidebar a:hover {
  color: gray;
}

a:active {
  color: gray;
}

.Sidebar {
  background-color: rgb(46, 46, 46) !important;
}
