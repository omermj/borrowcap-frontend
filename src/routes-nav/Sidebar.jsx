import { Container, Row, Col } from "react-bootstrap";
import UserContext from "../auth/UserContext";
import { useContext } from "react";
import { Link } from "react-router-dom";
import "./Sidebar.css";

/** Sidebar Component */

const Sidebar = () => {
  const { currentUser } = useContext(UserContext);

  const borrowerSidebar = () => {
    return (
      <Col className="Sidebar col-auto col-md-3 col-xl-2 px-sm-2 px-0 bg-dark">
        <div className="d-flex flex-column align-items-center align-items-sm-start px-3 pt-2 text-white min-vh-100">
          <ul className="nav nav-pills flex-column mb-sm-auto mb-0 align-items-center align-items-sm-start">
            <li className="nav-item">
              <Link to="/" className="nav-link align-middle px-0">
                <i className="fs-5 bi-house"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Home</span>
              </Link>
            </li>
            <li>
              <Link to="/borrower/apply" className="nav-link px-0 align-middle">
                <i className="fs-5 bi-coin"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Apply</span>
              </Link>
            </li>
            <li>
              <Link
                to="#submenu1"
                data-bs-toggle="collapse"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-4 bi-table"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Applications</span>{" "}
              </Link>
              <ul
                className="collapse show nav flex-column ms-1 sidebar-collapse-transition"
                id="submenu1"
                data-bs-parent="#menu"
              >
                <li className="w-100">
                  <Link to="/borrower/activerequests" className="nav-link px-0">
                    <i className="fs-5 bi-file-spreadsheet"></i>{" "}
                    <span className="d-none d-sm-inline">Requests</span>{" "}
                  </Link>
                </li>
                <li>
                  <Link to="/borrower/approvedloans" className="nav-link px-0">
                    <i className="fs-5 bi-list-check"></i>{" "}
                    <span className="d-none d-sm-inline">Approved</span>{" "}
                  </Link>
                </li>
              </ul>
            </li>
            <li>
              <Link
                to="/borrower/fundedloans"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-cash-coin"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Loans</span>
              </Link>
            </li>
            <li>
              <Link
                to="/borrower/wallet"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-wallet-fill"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Wallet</span>
              </Link>
            </li>
          </ul>
        </div>
      </Col>
    );
  };

  const investorSidebar = () => {
    return (
      <Col className="Sidebar col-auto col-md-3 col-xl-2 px-sm-2 px-0 bg-dark">
        <div className="d-flex flex-column align-items-center align-items-sm-start px-3 pt-2 text-white min-vh-100">
          <ul className="nav nav-pills flex-column mb-sm-auto mb-0 align-items-center align-items-sm-start">
            <li className="nav-item">
              <Link to="/" className="nav-link align-middle px-0">
                <i className="fs-5 bi-house"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Home</span>
              </Link>
            </li>
            <li>
              <Link
                to="/investor/availableinvestments"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-coin"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Invest</span>
              </Link>
            </li>
            <li>
              <Link
                to="/investor/pledges"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-clipboard-check-fill"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Pledges</span>
              </Link>
            </li>
            <li>
              <Link
                to="/investor/activeinvestments"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-cash-coin"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Investments</span>
              </Link>
            </li>
            <li>
              <Link
                to="/investor/wallet"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-wallet-fill"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Wallet</span>
              </Link>
            </li>
          </ul>
        </div>
      </Col>
    );
  };

  const underwriterSidebar = () => {
    return (
      <Col className="Sidebar col-auto col-md-3 col-xl-2 px-sm-2 px-0 bg-dark">
        <div className="d-flex flex-column align-items-center align-items-sm-start px-3 pt-2 text-white min-vh-100">
          <ul className="nav nav-pills flex-column mb-sm-auto mb-0 align-items-center align-items-sm-start">
            <li className="nav-item">
              <Link to="/" className="nav-link align-middle px-0">
                <i className="fs-5 bi-house"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Home</span>
              </Link>
            </li>
            <li>
              <Link
                to="/underwriter/requests"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-cash"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Requests</span>
              </Link>
            </li>
            <li>
              <Link
                to="/underwriter/approved"
                className="nav-link px-0 align-middle"
              >
                <i className="fs-5 bi-clipboard-check-fill"></i>{" "}
                <span className="ms-1 d-none d-sm-inline">Approved</span>
              </Link>
            </li>
          </ul>
        </div>
      </Col>
    );
  };

  const loggedOutSidebar = () => {
    return null;
  };

  let sidebarToDisplay = null;
  if (currentUser) {
    if (currentUser.roles.includes("admin"))
      sidebarToDisplay = underwriterSidebar;
    else if (currentUser.roles.includes("borrower"))
      sidebarToDisplay = borrowerSidebar;
    else if (currentUser.roles.includes("investor"))
      sidebarToDisplay = investorSidebar;
  } else sidebarToDisplay = () => <div></div>;

  return sidebarToDisplay();
};

export default Sidebar;
