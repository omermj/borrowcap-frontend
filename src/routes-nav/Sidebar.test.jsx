import { render } from "@testing-library/react";
import Sidebar from "./Sidebar";
import { BrowserRouter } from "react-router-dom";
import UserContext from "../auth/UserContext";

// smoke test
it("renders without crashing", () => {
  render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: { username: "test", roles: ["borrower"] } }}
      >
        <Sidebar />
      </UserContext.Provider>
    </BrowserRouter>
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: { username: "test", roles: ["borrower"] } }}
      >
        <Sidebar />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
