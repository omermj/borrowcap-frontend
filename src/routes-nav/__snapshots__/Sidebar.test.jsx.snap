// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Sidebar col-auto col-md-3 col-xl-2 px-sm-2 px-0 bg-dark col"
  >
    <div
      class="d-flex flex-column align-items-center align-items-sm-start px-3 pt-2 text-white min-vh-100"
    >
      <ul
        class="nav nav-pills flex-column mb-sm-auto mb-0 align-items-center align-items-sm-start"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link align-middle px-0"
            href="/"
          >
            <i
              class="fs-5 bi-house"
            />
             
            <span
              class="ms-1 d-none d-sm-inline"
            >
              Home
            </span>
          </a>
        </li>
        <li>
          <a
            class="nav-link px-0 align-middle"
            href="/borrower/apply"
          >
            <i
              class="fs-5 bi-coin"
            />
             
            <span
              class="ms-1 d-none d-sm-inline"
            >
              Apply
            </span>
          </a>
        </li>
        <li>
          <a
            class="nav-link px-0 align-middle"
            data-bs-toggle="collapse"
            href="/#submenu1"
          >
            <i
              class="fs-4 bi-table"
            />
             
            <span
              class="ms-1 d-none d-sm-inline"
            >
              Applications
            </span>
             
          </a>
          <ul
            class="collapse show nav flex-column ms-1 sidebar-collapse-transition"
            data-bs-parent="#menu"
            id="submenu1"
          >
            <li
              class="w-100"
            >
              <a
                class="nav-link px-0"
                href="/borrower/activerequests"
              >
                <i
                  class="fs-5 bi-file-spreadsheet"
                />
                 
                <span
                  class="d-none d-sm-inline"
                >
                  Requests
                </span>
                 
              </a>
            </li>
            <li>
              <a
                class="nav-link px-0"
                href="/borrower/approvedloans"
              >
                <i
                  class="fs-5 bi-list-check"
                />
                 
                <span
                  class="d-none d-sm-inline"
                >
                  Approved
                </span>
                 
              </a>
            </li>
          </ul>
        </li>
        <li>
          <a
            class="nav-link px-0 align-middle"
            href="/borrower/fundedloans"
          >
            <i
              class="fs-5 bi-cash-coin"
            />
             
            <span
              class="ms-1 d-none d-sm-inline"
            >
              Loans
            </span>
          </a>
        </li>
        <li>
          <a
            class="nav-link px-0 align-middle"
            href="/borrower/wallet"
          >
            <i
              class="fs-5 bi-wallet-fill"
            />
             
            <span
              class="ms-1 d-none d-sm-inline"
            >
              Wallet
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
